rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ds/DataSpaceClient.java

```diff
@@ -242,21 +251,30 @@ public boolean download(IRemoteSource source, ILocalDestination destination)
 
     @Override
     public RemoteSpace getGlobalSpace() {
-        throw new UnsupportedOperationException();
+        return new RestRemoteSpace(Dataspace.GLOBAL);
     }
 
     @Override
     public RemoteSpace getUserSpace() {
-        throw new UnsupportedOperationException();
+        return new RestRemoteSpace(Dataspace.USER);
     }
 
     @Override
-    public ListFile list(IRemoteSource source) throws NotConnectedException, PermissionException {
+    public ListFile list(IRemoteSource source, boolean recursive) throws NotConnectedException, PermissionException {
         StringBuffer uriTmpl = (new StringBuffer()).append(restDataspaceUrl).append(
                 source.getDataspace().value());
         ResteasyClient client = new ResteasyClientBuilder().httpEngine(httpEngine).build();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "client" which hides the field declared at line 83. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AHiddenFieldCheck)